src/Network/Network/AzureFirewall/NetworkRule/NewAzureFirewallNetworkRuleCommand.cs

```diff
@@ -76,6 +85,12 @@ public override void Execute()
         {
             base.Execute();
 
+            // One of SourceAddress or SourceIpGroup must be present
+            if ((SourceAddress == null) && (SourceIpGroup == null))
```

Comment:
SourceAddress is mandatory, you are trying to throw exception when SourceIpGroup is null, are you trying to have another parameter set which requires SourceIpGroup mandatory?

Reply:
Fixed to make SourceAddress optional. This check is to flag when both fields are null